src/main/drivers/pwm_rx.h

```diff
@@ -17,6 +17,8 @@
 
 #pragma once
 
+#include "drivers/timer.h"
```

Comment:
Well, not sure I like fwd declarations. You should not declare someone else's symbols, you should include their declarations and defs, and use them. In the IWYU way.
In this case, what if timer.c/.h is refactored with some name changes? Only noticed at linkage time using fwd decl.